planner/planner.go

```diff
@@ -618,17 +617,17 @@ func (p *MyPlanner) getPlanDetails(ctx *gin.Context) {
 	utils.LogErrorWithLevel(p.TripHTMLTemplate.Execute(ctx.Writer, tripResp), utils.LogError)
 }
 
-func (p *MyPlanner) asyncGetTripRespPlaceDetails(wg *sync.WaitGroup, resp *PlaceDetailsResp, place POI.Place) {
-	*resp = p.getTripFromPlace(place)
+func (p *MyPlanner) asyncGetTripRespPlaceDetails(ctx context.Context, wg *sync.WaitGroup, resp *PlaceDetailsResp, place POI.Place) {
+	*resp = p.getTripFromPlace(ctx, place)
 	wg.Done()
 }
 
-func (p *MyPlanner) getTripFromPlace(place POI.Place) PlaceDetailsResp {
+func (p *MyPlanner) getTripFromPlace(ctx context.Context, place POI.Place) PlaceDetailsResp {
```

Comment:
The function needs to return an error as well.

Reply:
This comment depends on whether we decide on the function signature `GetPhotoURL()`.

Reply:
If we decide that GetPhotoURL() from PhotoClient handles error cases internally and always return some valid photo data, we probably don't need to return an error in this function.

Reply:
This function should handle the error and provide the response with a default URL if needed

Reply:
Done, see [4d92fba](https://github.com/timwangmusic/Vacation-Planner/pull/338/commits/4d92fbacbfcac0d0847f62a05acaa06e0f8143d6)